### ISSUE TITLE
Pin TV shows to start screen

### ISSUE THREAD

**Issue by brinox** [akshay2000/XBMCRemoteRT#40]
Most of the time, I use Kodi and the remote to watch the next episode of a certain series. The process is as following:

1. Open the app
2. Swipe to TV shows
3. Tap *see all shows*
4. Wait for all shows to load, select TV show
5. Swipe to episodes
6. Select episode and play

That takes some time, especially point 4, as my library has grown quite a bit. So how about enabling the user to pin a certain TV show to the start screen so the app automatically launches and shows the page from step 5 / 6? One could use TV show fanart for the tile's background. 

A smiliar feature is offered by [Rico Suter's app](http://www.windowsphone.com/en-us/store/app/xbmc-remote/64f736e7-df4a-414d-b51e-f21e22698bab).

**Comment by akshay2000** [comment]
So this show specific tile is specific only to show or is it specific to the server too?

In other word, what happens in following scenario:
1. User has two servers: A and B.
2. User connects to server A.
3. User pins show X to start screen.
4. User connects to server B. Now, the last successfully connected server is B.
5. OS terminates the app while user is busy watching some cat videos on MetroTube.
6. User taps the secondary tile.

Here, two things could happen:
1. App attempts to connect to server A (since that was the server tile was created with).
2. App attempts to connect to server B (since that was recently connected server).

Which one should the app do? I'm leaning towards the second one since one tile would suffice for launching the show on all servers.

**Comment by brinox** [comment]
I don't know the users' habits, personally, I have 3 profiles in saved in my app.
1. livingroom
2. bedroom
3. friend's home

The first two Kodi instances have their library synced, because they rely on the same MySQL database. If this is the case, the second suggestion of your's would be best, as the ID for the TV show is the same on both instances. Moreover, the speech control feature works by the same mechanism by always using the most recently used profile, doesn't it?

On the other hand, if one has two Kodi instances running, that have their libraries based on the very same file share but are kept in different SQLite files, the ID for the same show would be different on both devices (right?). As I am assuming that you would save the show's ID in the tile, wouldn't that cause any difficulties?

If I return from a movie session at my friend's and now want to continue to watch a show at home, the app tries to connect to the third profile, fails with a timeout. Then, I have to connect to my livingroom server manually and then a) select the show by hand or b) terminate / freeze the app and tap the tile on my start screen. Given this szenario, the first suggestion would fit better.

tl;dr: I personally prefer the second suggestion, too, as this would cover most situations. However, I wonder if the assumption of equal IDs in two instances' libraries is true in most people's setups. 

I actually wonder how many people actually use more than one profile... :)

**Comment by gabrielsadaka** [comment]
You wouldn't be using the ID to find the show to play because as you said that can change depending on the server, the best way would be to search by the show title because that should be unique and if it isn't it could revert to the search screen showing you all of the matching results. 

I agree with Akshay, it should try to automatically connect to the last connected server but if that fails it could direct you to the server selection screen then take you back to the show. If the server you chose doesn't have the show it would then redirect to the main screen as there really isn't much else that can be done.

**Comment by brinox** [comment]
I totally agree with @gabrielsadaka, that behavior would be awesome.

**Comment by akshay2000** [comment]
@brinox Good thing, it's already implemented that way. It's just taking a lot of time because of tight schedule.

**Comment by brinox** [comment]
Nice, I'll close this issue as soon as the feature hits beta and I can test it ;)

**Comment by akshay2000** [comment]
No problem. Issue will close itself as soon as the commits get merged into master. I'll release the beta based on dev. If no problem is seen, I'll merge into master.

**Comment by akshay2000** [comment]
@brinox Can you report back? Is this feature ready for release?

**Comment by brinox** [comment]
Everything seems fine, works like a charm :+1:
